### DISCUSSION TITLE
jsregexp

### DISCUSSION THREAD

**Discussion by nxmxsxs** [L3MON4D3/LuaSnip#741] [General]
Hello. I've installed jsregexp on nixos via nix package manager (which basically pulls directly from luarocks) but I've noticed that it's [a few commits behind](https://github.com/kmarius/jsregexp/compare/kmarius:dd65498ae2c29b882d6c02c0a30577b08d660b94...c3e473240eebb65a8870abebafeff83b6c9e7f16) compared to luasnip's bundled version. Are there any functional differences between the two?

**Comment by L3MON4D3** [comment]
IIRC, the only breaking difference is that an unsuccessful `compile` throws an error, but that should be it (@kmarius: right?)

**Reply by kmarius** [reply]
No, there is no functional difference. The changes to `compile`/`compile_safe` were introduced after the pinned commit in the bundled version (and therefore after the nix/luarocks release). The commits linked above are laying the groundwork for the (ongoing..) implementation of the javascript RegExp api.
